theSrc/scripts/PlotColors.es6.js

```diff
@@ -0,0 +1,62 @@
+
+// NB local require statements cannot be enabled until the ES6 porting is complete
+// const Utils = require('./utils/Utils.es6')
+/* global Utils */
+
+/* To Refactor:
+ *   * the constuctor of PlotColors has a side effect that creates the legendGroup used by PlotData etc
+ *
+ */
+
+// This class assigns colors to the plot data
+// If group array is supplied, it will couple the array with the color wheel
+// If no group array is supplied, colors are rotated through the color wheel
+class PlotColors {
+  constructor(plotData) {
+    this.plotData = plotData;
+    this.plotData.legendGroups = [];
+
+    this.groupToColorMap = {};
+
```

Comment:
i rerwrote this loop. In future I would like to pull out the side effect part, but for now just left it as a note to refactor